my-app/src/Hooks/useAuth.tsx

```diff
@@ -0,0 +1,38 @@
+import React, { createContext, useContext, useState, useEffect } from "react";
+import { useNavigate } from "react-router-dom";
+
+import { authService } from "../Services/AuthService";
+import { IUser } from "../Types/user";
+
+const AuthContext: any = createContext(null);
+
+export function AuthProvider({ children }: { children: React.ReactNode }) {
+  const navigate = useNavigate();
+  const userToken = localStorage.getItem("user");
+  const userData = userToken ? JSON.parse(userToken) : {};
```

Comment:
We do not need userToken at all? 
```suggestion
  const userData = localStorage.getItem("user") ? JSON.parse(userToken) : {};
```